lec02-basic-r.Rmd

```diff
@@ -1,4 +1,765 @@
+double), integer, and logical. But what happens if we try to mix these types in 
+>   a single vector? Find out by using `class` to test these examples.
+
+> ```r
+> num_char <- c(1, 2, 3, 'a')
+> num_logical <- c(1, 2, 3, TRUE)
+> char_logical <- c('a', 'b', 'c', TRUE)
+> tricky <- c(1, 2, 3, '4')
+> ```
+
+> <!-- * _Answer_: R implicitly converts them to all be the same type -->
+>
+
+> - This happens because vectors can be of only one data type. Instead of
+>  throwing an error and saying that you are trying to mix different types in
+>  the same vector, R tries to convert (coerce) the content of this vector to
+>  find a "common denominator". A logical can be turn into 1 or 0, and a number
+>  can be turned into a string/character representation. It would be difficult
+>  to do it the other way around: would 5 be TRUE or FALSE? What number would
+>  't' be?
+
+> - In R, we
+>   call converting objects from one class into another class
+>   _coercion_. These conversions happen according to a hierarchy,
+>   whereby some types get preferentially coerced into other
+>   types. Can you draw a diagram that represents the hierarchy of how
+>   these data types are coerced?
+> <!-- * _Answer_: `logical -> numeric -> character <-- logical` -->
+
+
+```{r, echo=FALSE, eval=FALSE}
+num_char <- c(1, 2, 3, "a")
+
+num_logical <- c(1, 2, 3, TRUE)
+
+char_logical <- c("a", "b", "c", TRUE)
+
+tricky <- c(1, 2, 3, "4")
+```
+
+
+## Subsetting vectors
+
+If we want to extract one or several values from a vector, we must provide one
+or several indices in square brackets. For instance:
+
+```{r}
+animals <- c("mouse", "rat", "dog", "cat")
+animals[2]
+animals[c(3, 2)]
+```
+
+We can also repeat the indices to create an object with more elements than the
+original one:
+
+```{r}
+more_animals <- animals[c(1, 2, 3, 2, 1, 4)]
+more_animals
+```
+
+R indices start at 1. Programming languages like Fortran, MATLAB, Julia, and R
+start counting at 1, because that's what human beings typically do. Languages in
+the C family (including C++, Java, Perl, and Python) count from 0 because that's
+simpler for computers to do.
+
+
+### Conditional subsetting
+
+Another common way of subsetting is by using a logical vector. `TRUE` will
+select the element with the same index, while `FALSE` will not:
+
+```{r}
+weight_g <- c(21, 34, 39, 54, 55)
+weight_g[c(TRUE, FALSE, TRUE, TRUE, FALSE)]
+```
+
+Typically, these logical vectors are not typed by hand, but are the output of
+other functions or logical tests. For instance, if you wanted to select only the
+values above 50:
+
+```{r}
+weight_g > 50    # will return logicals with TRUE for the indices that meet the condition
+## so we can use this to select only the values above 50
+weight_g[weight_g > 50]
+```
+
+You can combine multiple tests using `&` (both conditions are true, AND) or `|`
+(at least one of the conditions is true, OR), similar to how you can do when
+using a search engine online:
```

Comment:
With Google becoming "smart", I'm not sure if this is something you really do with search engines any more.